client/thin-replica-client/src/grpc_connection.cpp

```diff
@@ -133,13 +149,12 @@ GrpcConnection::Result GrpcConnection::openDataStream(const SubscriptionRequest&
 
   ConcordAssert(status == future_status::ready);
   data_stream_ = stream.get();
-  if (data_stream_) return Result::kSuccess;
+  if (!data_stream_) {
+    data_context_.reset();
+    return Result::kFailure;
+  }
```

Comment:
If `data_stream_` exists, I think we should call `Finish()` on the stream to get the rpc status. If the status is ok, we return `Result::kSuccess`, otherwise, we must check for `OUT_OF_RANGE` and `NOT_FOUND` errors, and return the corresponding result, else return `Result::kFailure`. This is because we must catch and relay `OUT_OF_RANGE` and `NOT_FOUND` errors from the TRS to the concord client.
Same comment for `openHashStream` method.

Reply:
thanks, @shgandhi. will do this change.

Reply:
It looks like you reverted your change. 🤔  
Maybe we can fix this separately.